tests/integration/awslambda/test_lambda_api.py

```diff
@@ -2829,14 +2828,122 @@ def test_code_signing_not_found_excs(
+
+        "AccountLimit": {
+            "TotalCodeSize": 80530636800,
+            "CodeSizeUnzipped": 262144000,
+            "CodeSizeZipped": 52428800,
+            "ConcurrentExecutions": 10,
+            "UnreservedConcurrentExecutions": 10
+        }"""
         acc_settings = lambda_client.get_account_settings()
-        acc_settings["AccountLimit"] = sorted(list(acc_settings["AccountLimit"].keys()))
-        acc_settings["AccountUsage"] = sorted(list(acc_settings["AccountUsage"].keys()))
-        snapshot.match("acc_settings_modded", acc_settings)
+        acc_settings_modded = acc_settings
+        acc_settings_modded["AccountLimit"] = sorted(list(acc_settings["AccountLimit"].keys()))
+        acc_settings_modded["AccountUsage"] = sorted(list(acc_settings["AccountUsage"].keys()))
+        snapshot.match("acc_settings_modded", acc_settings_modded)
+
+    @pytest.mark.aws_validated
+    def test_account_settings_total_code_size(
+        self, lambda_client, create_lambda_function, dummylayer, cleanups, snapshot
+    ):
+        """Caveat: Could be flaky if another test simultaneously deletes a larger lambda function in the same region.
+        Hence, testing for monotonically increasing `TotalCodeSize` rather than matching exact differences.
+        """
+        acc_settings0 = lambda_client.get_account_settings()
+
+        # 1) create a new function
+        function_name = f"lambda_func-{short_uid()}"
+        create_lambda_function(
+            handler_file=TEST_LAMBDA_PYTHON_ECHO,
+            func_name=function_name,
+            runtime=Runtime.python3_9,
+        )
+        acc_settings1 = lambda_client.get_account_settings()
+        assert (
+            acc_settings1["AccountUsage"]["TotalCodeSize"]
+            > acc_settings0["AccountUsage"]["TotalCodeSize"]
+        )
+        assert (
+            acc_settings1["AccountUsage"]["FunctionCount"]
+            > acc_settings0["AccountUsage"]["FunctionCount"]
+        )
+        snapshot.match(
+            "total_code_size_diff_create_function",
+            acc_settings1["AccountUsage"]["TotalCodeSize"]
+            - acc_settings0["AccountUsage"]["TotalCodeSize"],
+        )
+
+        # 2) update the function
+        update_zip_file = create_lambda_archive(
+            load_file(TEST_LAMBDA_PYTHON_ECHO), get_content=True
+        )
+        lambda_client.update_function_code(
+            FunctionName=function_name, ZipFile=update_zip_file, Publish=True
+        )
+        acc_settings2 = lambda_client.get_account_settings()
+        assert (
+            acc_settings2["AccountUsage"]["TotalCodeSize"]
+            > acc_settings1["AccountUsage"]["TotalCodeSize"]
+        )
+        snapshot.match(
+            "total_code_size_diff_update_function",
+            acc_settings2["AccountUsage"]["TotalCodeSize"]
+            - acc_settings1["AccountUsage"]["TotalCodeSize"],
+        )
```

Comment:
Can you also add a test case where we create a function, publish it as a version, and then change something which is not the code, like the runtime? Would be interesting how it is counted then.

Reply:
great suggestion. I added an appropriate test case `test_account_settings_total_code_size_config_update` that particularly covers the config update => publish cycle.